halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/SpinnakerPublicService.java

```diff
@@ -25,18 +25,25 @@
 @Data
 abstract public class SpinnakerPublicService<T> extends SpinnakerService<T> {
   String protocol = "http";
+  String domain;
 
   public abstract String getPublicAddress();
 
   public String getPublicEndpoint() {
-    return getProtocol() + "://" + getPublicAddress() + ":" + getPort();
+    return getProtocol() + "://" + domain + ":" + getPort();
```

Comment:
At some point, would like to get away from doing all of this string-work manually.